pkg/pluginloader/loader.go

```diff
@@ -0,0 +1,84 @@
+package pluginloader
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/spf13/cobra"
+)
+
+// TODO:
+// Unmarshal the YAML
+// Extract the commands using plugin.go (command structure)
+
+// Create a Cobra Command
+func createCommand(cmdStruct *CommandConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     cmdStruct.Name,
+		Short:   cmdStruct.ShortDescription,
+		Example: cmdStruct.Examples,
+		Args:    cobra.ExactArgs(len(cmdStruct.MapsTo.Args)),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			args = append([]string{cmdStruct.MapsTo.Subcommand}, args...)
+			c := exec.Command(cmdStruct.MapsTo.Name, args...)

```

Comment:
I would preffer us to start with calling  method properly. Then we can inject different ways to exec commands that can use os.exec as well. This is called #premature-specification and affected your entire POC/Architecture.

This way arguments are not standardized, secondly we do not have ability to strongly type them on the plugin.
Let me know if you know what I mean as this is hard to explain 

TL;DR - let's not execute external process and call function
- Let's see if we can specify arguments in the way that they are clearly defined on plugin

Reply:
Calling external CLI is a very cool although a commonly resolved problem in the golang ecosystem.
I do not want to steal cool feature from you and we can keep this as capability, but I did not think that people will use charmil for that where you can have this:

https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/plugin/plugin.go

https://github.com/openshift-online/ocm-cli/blob/master/pkg/plugin/plugin.go

https://github.com/openshift/odo/blob/main/pkg/odo/cli/plugins/plugin_handler.go